bundles/specmate-integration-test/src/com/specmate/test/integration/CrudTest.java

```diff
@@ -937,4 +938,128 @@ public void testDuplicate() {
 		// payload.getJSONObject(1), true));
 
 	}
+
+	@Test
+	public void testRecycleCEG() {
+		JSONObject folder = postFolderToTopFolder();
+		String folderName = getId(folder);
+
+		JSONObject requirement = postRequirement(folderName);
+		String requirementName = getId(requirement);
+
+		JSONObject ceg = postCEG(folderName, requirementName);
+		String cegName = getId(ceg);
+
+		JSONObject testSpec = postTestSpecification(folderName, requirementName, cegName);
+		String testSpecName = getId(testSpec);
+
+		JSONObject testSpec2 = postTestSpecification(folderName, requirementName, cegName);
+		String testSpecName2 = getId(testSpec2);
+
+		JSONObject process = postProcess(folderName, requirementName);
+		String processName = getId(process);
+
+		JSONObject testSpec3 = postTestSpecification(folderName, requirementName, processName);
+		String testSpecName3 = getId(testSpec3);
+
+		recycleObject(folderName, requirementName, cegName);
+
+		JSONObject retrievedFolder = getObject(folderName);
+		Assert.assertTrue(retrievedFolder.getBoolean(BasePackage.Literals.IRECYCLED__HAS_RECYCLED_CHILDREN.getName()) == true);
+		Assert.assertTrue(retrievedFolder.getBoolean(BasePackage.Literals.IRECYCLED__RECYCLED.getName()) == false);

```

Comment:
Since you retrieve a boolean, there is no need to compare to true. Change to
`Assert.assertTrue(retrievedFolder.getBoolean(BasePackage.Literals.IRECYCLED__HAS_RECYCLED_CHILDREN.getName()));`